### ISSUE TITLE
Allocation rules of training set validation set and test set?  :)

### ISSUE THREAD

**Issue by lisherlock** [wasserth/TotalSegmentator#28]
Hi! Thanks for your great work of the total body segmentation! It's really meaningful. But I have a few questions about your division of training set, verification set and test set.
I noticed that you said in your paper about it is 'The dataset of 1204 subjects was randomly split into 1082 training subjects (89.9%), 57 validation subjects (4.7%) and 65 subjects for final testing (5.4%)'. But when you trained nnUNET, I noticed that the parameter of 'fold' is '0'. Does that mean that your verification set is 1139/5 instead of 57 cases during the training? Or did you change the 'fold' parameter from '0' to 'all' and retrained? 
Looking forward to your reply!

**Comment by lisherlock** [comment]
> I split the 1139 subjects into 20 folds and trained/validated only on the first fold. This resulted in 1082 training and 57 validation subjects.

Thank you very much! I have already got it.

**Comment by naayem** [comment]
@wasserth Hello, Can you share your split file?

**Comment by wasserth** [comment]
The exact split of the dataset can be found in the file meta.csv inside of the [dataset](https://doi.org/10.5281/zenodo.6802613).

For more details see here: https://github.com/wasserth/TotalSegmentator#train--validation--test-split

**Comment by naayem** [comment]
Hello, Thanks for the answer. I achieved it. So if anyone wants to replicate some particular splits, it is well explained in here:
[https://github.com/MIC-DKFZ/nnUNet/blob/master/documentation/common_questions.md#creating-and-managing-data-splits](url)
Then to replicate the particular split of Totalsegmentor paper it is explained in more details in here:


> I split the 1139 subjects into 20 folds and trained/validated only on the first fold. This resulted in 1082 training and 57 validation subjects.

So no validation folder. nnUnet sort it automatically with a split file of 20 folds. Just separate the test set in another folder and keep both training and validation set in the training folder. Then from meta.csv sort the cases 'sXXX' corresponding to training and the cases corresponding to validation. (val_case1 = corresponding 'sXXX')
In this particular example the splits_final.pkl file contains the following data structure:
splits = [
    {'train': ['tr_case1', 'tr_case2',..., 'tr_case1082'], 'val': ['val_case1', 'val_case57']},
    {'train': [], 'val': []},
    {'train': [], 'val': []},
    {'train': [], 'val': [},
    {'train': [], 'val': []}
]
Use load_pickle and save_pickle from batchgenerators.utilities.file_and_folder_operations for loading/storing the splits.
